src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,186 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as mssql from 'vscode-mssql';
+import { defaultBindingResult, defaultSqlBindingTextLines, sqlBindingResult } from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+import { AzureFunctionsService } from '../services/azureFunctionsService';
+import * as azureFunctionUtils from '../azureFunction/azureFunctionUtils';
+import * as constants from '../constants/constants';
+import * as path from 'path';
+import { executeCommand, generateQuotedFullName } from '../utils/utils';
+
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await azureFunctionUtils.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		let projectFile = await isAzureFunctionProjectOpen();
+		if (!projectFile) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = getNewFunctionFile(projectFile);
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: constants.functionNameTitle,
+			value: table,
+			ignoreFocusOut: true
+		});
+		if (!functionName) {
+			return;
+		}
+
+		// create C# HttpTrigger
+		await azureFunctionApi.createFunction({
+			language: 'C#',
+			templateId: 'HttpTrigger',
+			functionName: functionName,
+			folderPath: projectFile
+		});
+
+		// once steps are addressed we add SQL Bindings Nuget and Connection String to local.settings.json
```

Comment:
What's the "once steps are addressed" for? Typically comments describe what's happening below so that wouldn't be necessary here.

(although honestly this whole comment doesn't super necessary - the function names already indicate what's being done pretty clearly)